unittest/gunit/keyring_vault/vault_environment.h

```diff
@@ -0,0 +1,90 @@
+#ifndef MYSQL_GUNIT_VAULT_ENVIRONMENT_H
+#define MYSQL_GUNIT_VAULT_ENVIRONMENT_H
+
+#include <string>
+
+#include <gtest/gtest.h>
+
+class Vault_environment : public ::testing::Environment {
+ public:
+  Vault_environment();
+
+  virtual ~Vault_environment() {}
+
+  const std::string &get_uuid() const { return uuid_; }
+  const std::string &get_key1_id() const { return key1_id_; }
+  const std::string &get_key2_id() const { return key2_id_; }
+
+  const char *get_uuid_raw() const { return uuid_.c_str(); }
+  const char *get_key1_id_raw() const { return key1_id_.c_str(); }
+  const char *get_key2_id_raw() const { return key2_id_.c_str(); }
+
+  const std::string &get_default_conf_file_name() const
+  {
+    return default_conf_file_name_;
+  }
+  const std::string &get_invalid_conf_file_name() const
+  {
+    return invalid_conf_file_name_;
+  }
+  const std::string &get_non_existing_conf_file_name() const
+  {
+    return non_existing_conf_file_name_;
+  }
+
+  const std::string &get_mount_point_path() const
+  {
+    return mount_point_path_;
+  }
+
+  const std::string &get_admin_token() const { return admin_token_; }
+
+
+  static Vault_environment *create_instance()
+  {
+    if (instance_ != NULL)
+      return NULL;
+
+    instance_= new Vault_environment();
+    return instance_;
+  }
+  static Vault_environment *get_instance() { return instance_; }
```

Comment:
Can we just extend this method to create instance when needed and remove create_instance() at all?

Reply:
Not really, the problem is that once an instance of `Vault_environment` is passed to `::testing::AddGlobalTestEnvironment()`
```c++
::testing::AddGlobalTestEnvironment(Vault_environment::create_instance());
```
we are no longer responsible for its lifetime - Google Test environment is supposed to free this resource.
Therefore, this is not a canonical `singleton` pattern.